ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -55,6 +56,7 @@ func NewService(context *context.Context) entity.ProjectEntity {
 		cache:          entity.SetupTaskDefinitionCache(),
 		projectContext: context,
 		timeSleeper:    &utils.TimeSleeper{},
+		taskRoleArn:    context.TaskRoleArn,
```

Comment:
TaskRoleArn is used when creating task definition, not used for create service hence doesn't make sense to be part of the service object.